scripts/solve_network.py

```diff
@@ -1055,14 +1055,14 @@ def extra_functionality(n, snapshots):
     if config["sector"]["enhanced_geothermal"]["enable"]:
         add_flexible_egs_constraint(n)
 
-    if snakemake.params.custom_extra_functionality:
-        source_path = snakemake.params.custom_extra_functionality
+    if params.custom_extra_functionality:
+        source_path = params.custom_extra_functionality
         assert os.path.exists(source_path), f"{source_path} does not exist"
         sys.path.append(os.path.dirname(source_path))
         module_name = os.path.splitext(os.path.basename(source_path))[0]
         module = importlib.import_module(module_name)
         custom_extra_functionality = getattr(module, module_name)
-        custom_extra_functionality(n, snapshots, snakemake)
+        custom_extra_functionality(n, snapshots)
```

Comment:
this is a better and cleaner approach, but note @fneum @lindnemi that this requires the ariadne code to remove dependency on the snakemake object in here https://github.com/PyPSA/pypsa-ariadne/blob/1eef9c6a9dba28a9bd286a01661089c28ab134e5/workflow/scripts/additional_functionality.py#L527

Reply:
@lindnemi, do you need the snakemake object or is it ok to remove here?